### ISSUE TITLE
runtime-dyno-metadata warning even if SENTRY_DSN is not set

### ISSUE THREAD

**Issue by jonekdahl** [getsentry/sentry-ruby#885]
We're using Sentry in our app that runs on Heroku, but we don't install the Sentry add-on for our [review apps](https://devcenter.heroku.com/articles/github-integration-review-apps), so SENTRY_DSN is not set for those apps.

I think I've read somewhere that one way to "deactivate" Sentry is to simply not set SENTRY_DSN, so this should be fine. However, I still get the runtime-dyno-metadata warning:

```
WARN -- sentry: ** [Raven] You are running on Heroku but haven't enabled Dyno Metadata. For Sentry's release detection to work correctly, please run `heroku labs:enable runtime-dyno-metadata`
```

I would like Sentry to be completely silent when SENTRY_DSN is not available, or at least not log warnings.

**Comment by SampsonCrowley** [comment]
It would also be nice not to get this warning when the version is already explicitly set

**Comment by st0012** [comment]
@jonekdahl what environment will the review app have? will it be `production` too or something like `review` or `staging`?

**Comment by jonekdahl** [comment]
@st0012 Are you asking about RAILS_ENV? I don't believe Heroku puts any restrictions on that, so it could be any of the above. We're using 'production'.

**Comment by pil0u** [comment]
I am not sure if it's the right place to ask but on the very bottom of [this page from the docs](https://docs.sentry.io/platforms/ruby/configuration/options/), it says:
> With Sentry, you may **either** set the `SENTRY_DSN` environment variable (recommended), **or** set your DSN manually in a config block

When my `config.dns` is set in my Rails app and I properly ran the `heroku labs:enable runtime-dyno-metadata -a <app_name>` command, I still get the warning mentioned in this issue.
If on top of that, I set the `SENTRY_DSN` variable in my Heroku variables, the warning disappears.

From the documentation, I thought that one solution _or_ the other should work. Should I open another issue for that matter?

**Comment by st0012** [comment]
hey sorry for the long delay 🙏 I have fixed this in #1471 and it will be included in the next patch release.
(because `sentry-raven` has entered maintenance mode, the fix will not be backported there. you can follow the [migration guide](https://docs.sentry.io/platforms/ruby/migration/) to migrate to the new SDK 🙂)
